### PR TITLE
Test automation framework

### PR DESCRIPTION
**Pull request by LazarevaAleksandra**

### CODE REVIEW

File: Test_Automation_Framework/Test_Automation_Framework/Elements/BaseElement.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
+﻿using Epam_TestAutomation_Core.Browser;
+using Epam_TestAutomation_Core.Driver;
+using OpenQA.Selenium;
+using System.Collections.ObjectModel;
+using System.Drawing;
+
+namespace Epam_TestAutomation_Core.Elements
+{
+    public abstract class BaseElement : IBaseElement
```
Comment (MariaEroshevich) [inline Test_Automation_Framework/Test_Automation_Framework/Elements/BaseElement.cs:9]: think about exactly which methods you need for tests and remove the extra ones.

File: Test_Automation_Framework/Test_Automation_Framework/Browser/Browser.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
+﻿using OpenQA.Selenium.Interactions;
+using OpenQA.Selenium.Support.UI;
+using OpenQA.Selenium;
+using System.Collections.ObjectModel;
+using Epam_TestAutomation_Utilities.Logger;
+using Epam_TestAutomation_Core.Helper;
+
+namespace Epam_TestAutomation_Core.Browser
```
Comment (MariaEroshevich) [inline Test_Automation_Framework/Test_Automation_Framework/Browser/Browser.cs:9]: think about exactly which methods you need for tests and remove the extra ones.

File: Test_Automation_Framework/Test_Automation_Framework/Utils/Waiters.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public static class Waiters
+    {
+        public static void WaitForPageLoad() => BrowserFactory.Browser.Waiters().Until
+            (condition => BrowserFactory.Browser.ExecuteScript("return document.readyState").Equals("complete"));
+
+        public static void WaitForCondition(Func<bool> condition) => BrowserFactory.Browser.Waiters().Until(x => condition.Invoke());
+
+        public static void WaitSpinner()
+        {
```
Comment (MariaEroshevich) [inline Test_Automation_Framework/Test_Automation_Framework/Utils/Waiters.cs]: Is this locator working for epam.com site?

File: Test_Automation_Framework/EPAM_TestAutomation_Tests/EpamSiteNavigateTests.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+using Epam_TestAutomation_Core.Helper;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Epam_TestAutomation_Tests
+{
```
Comment (MariaEroshevich) [inline Test_Automation_Framework/EPAM_TestAutomation_Tests/EpamSiteNavigateTests.cs]: First of all you have to rewrite the tests you wrote earlier for Selenium lessons.
The second solution doesn't work. The problem is with config file and TestSettings.cs classes.

File: Test_Automation_Framework/Epam_TestAutomation_BusinessLogic/PageObjects/PageObjects.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿using Epam_TestAutomation_Core.BasePage;
+using OpenQA.Selenium;
+using Epam_TestAutomation_Core.Elements;
+using Epam_TestAutomation_Core.Browser;
+using Epam_TestAutomation_Core.Helper;
+
+namespace Epam_TestAutomation_BusinessLogic.PageObjects
+{
+    internal class PageObjects : BasePage
```
Comment (MariaEroshevich) [inline Test_Automation_Framework/Epam_TestAutomation_BusinessLogic/PageObjects/PageObjects.cs]: you should name the class as the site page name. For example, SearchResultsPage

File: Test_Automation_Framework/EPAM_TestAutomation_Tests/BaseTest.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            Logger.Info("Test begin");
+            BrowserFactory.Browser.GoToUrl(TestSettings.ApplicationUrl);
+            Waiters.WaitForPageLoad();
+        }
+
+        [TearDown]
+        public virtual void CleanTest()
+        {
+            Logger.Info("Test finish");
```
Comment (MariaEroshevich) [inline Test_Automation_Framework/EPAM_TestAutomation_Tests/BaseTest.cs:36]: Why didn't you add screenshoter?

**Review by MariaEroshevich — CHANGES_REQUESTED** [review]

File: Test_Automation_Framework/EPAM_TestAutomation_Tests/Advansed_EpamSiteNavigateTests.cs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+                "Česká Republika (Čeština)", 
+                "India (English)", 
+                "Україна (Українська)",
+                "Czech Republic (English)", 
+                "日本 (日本語)", 
+                "中国 (中文)", 
+                "DACH (Deutsch)", 
+                "Polska (Polski)"};
+            _searchResultsPage.LanguagesButton.Click();
+            var actualListOfLanguages = BrowserFactory.Browser.Waiters().
```
Comment (MariaEroshevich) [inline Test_Automation_Framework/EPAM_TestAutomation_Tests/Advansed_EpamSiteNavigateTests.cs]: move webElement to separate page object (Epam_TestAutomation_BusinessLogic.PageObjects.Pages)

File: Test_Automation_Framework/EPAM_TestAutomation_Tests/Basic2_EpamSiteNavigateTests.cs [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        }
+
+        [Test]
+        public void CheckTheListOfCountriesInTheCareerButtonTest()
+        {
+            var countries = new List<string> { "AMERICAS", "EMEA", "APAC" };
+
+            _searchResultsPage.CareerButton.Click();
+            var ourOfficesCareer = _searchResultsPage.OurOfficesCareer;
+            var careerElements = ourOfficesCareer.FindElements(By.XPath(".//div"))
```
Comment (MariaEroshevich) [inline Test_Automation_Framework/EPAM_TestAutomation_Tests/Basic2_EpamSiteNavigateTests.cs]: move webElement to the Epam_TestAutomation_BusinessLogic.PageObjects.Pages 
Fix for all tests

File: Test_Automation_Framework/Epam_TestAutomation_TestData/test.json [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (MariaEroshevich) [inline Test_Automation_Framework/Epam_TestAutomation_TestData/test.json]: why do you need this file?

**Review by MariaEroshevich — CHANGES_REQUESTED** [review]

File: Test_Automation_Framework/Epam_TestAutomation_TestData/test.json (hunk 9 above)
Comment (LazarevaAleksandra) [reply Test_Automation_Framework/Epam_TestAutomation_TestData/test.json]: forgot to delete))

File: Test_Automation_Framework/EPAM_TestAutomation_Tests/Basic2_EpamSiteNavigateTests.cs [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+{
+    [TestFixture]
+    public class Basic2_EpamSiteNavigateTests : BaseTest
+    {
+        private SearchResultsPage _searchResultsPage;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _searchResultsPage = new SearchResultsPage();
```
Comment (MariaEroshevich) [inline Test_Automation_Framework/EPAM_TestAutomation_Tests/Basic2_EpamSiteNavigateTests.cs:16]: Why do you use SearchResultsPage if you navigate to the MainPage?

File: Test_Automation_Framework/Test_Automation_Framework/Helper/UITestSettings.cs [hunk 11]
```diff
@@ -0,0 +1,20 @@
+﻿//using Epam_TestAutomation_Core.Enum;
```
Comment (MariaEroshevich) [inline Test_Automation_Framework/Test_Automation_Framework/Helper/UITestSettings.cs:1]: remove unnecessary files

**Review by MariaEroshevich — CHANGES_REQUESTED** [review]

File: Test_Automation_Framework/EPAM_TestAutomation_Tests/Basic2_EpamSiteNavigateTests.cs (hunk 10 above)
Comment (LazarevaAleksandra) [reply Test_Automation_Framework/EPAM_TestAutomation_Tests/Basic2_EpamSiteNavigateTests.cs:16]: Followed previous comment

Comment (MariaEroshevich) [reply Test_Automation_Framework/EPAM_TestAutomation_Tests/Basic2_EpamSiteNavigateTests.cs:16]: Yes, but as I can see your browser navigates to the epam.com site and then makes a search from first page and the navigates to the search results page (also for clicking to the careerButton). I mean you should create the separate classes for each page.

**Review by MariaEroshevich — CHANGES_REQUESTED** [review]

**Review by MariaEroshevich — APPROVED** [review]

**Pull request merged**